packages/@aws-cdk/aws-redshift/lib/parameter-group.ts

```diff
@@ -62,17 +124,44 @@ export class ClusterParameterGroup extends ClusterParameterGroupBase {
    */
   public readonly clusterParameterGroupName: string;
 
+  /**
+   * The parameters in the parameter group
+  */
+  readonly parameters: { [name: string]: string };
+
+
+  protected updatableParameters = true;
   constructor(scope: Construct, id: string, props: ClusterParameterGroupProps) {
     super(scope, id);
-
+    this.parameters = props.parameters;
     const resource = new CfnClusterParameterGroup(this, 'Resource', {
       description: props.description || 'Cluster parameter group for family redshift-1.0',
       parameterGroupFamily: 'redshift-1.0',
-      parameters: Object.entries(props.parameters).map(([name, value]) => {
-        return { parameterName: name, parameterValue: value };
-      }),
+      parameters: Lazy.any({ produce: () => this.parseParameters() }),
     });
 
     this.clusterParameterGroupName = resource.ref;
   }
+  private parseParameters(): any {
+    return Object.entries(this.parameters).map(([name, value]) => {
+      return { parameterName: name, parameterValue: value };
+    });
+  }
+
+  public addParameter(name: string, value: string): AddParameterResult {
+    if (this.updatableParameters) {
```

Comment:
Why is this check here? Is it not enough to just check if the `existingValue` is `undefined`?

Reply:
You're right, which also means we don't need `updatableParameters` at all. I'll update the PR for that